devtools/server/actors/replay/connection.js

```diff
@@ -1265,25 +1277,36 @@ async function fetchText(recordingId, url) {
   }
 
   try {
-    const response = await fetch(url, {
-      //
-      credentials: "include",
+    const channel = NetUtil.newChannel({
+      uri: urlObj.toString(),
+      loadingPrincipal: browser.contentPrincipal,
+      triggeringPrincipal: browser.contentPrincipal,
+      contentPolicyType: Ci.nsIContentPolicy.TYPE_DOCUMENT,
+      securityFlags: Ci.nsILoadInfo.SEC_ALLOW_CROSS_ORIGIN_SEC_CONTEXT_IS_NULL,
     });
-    if (response.status < 200 || response.status >= 300) {
-      console.error("Error fetching recording resource", url, response);
-      pingTelemetry("sourcemap-upload", "fetch-bad-status", {
-        message: `Request got status: ${response.status}`,
-        status: response.status,
-        url: ["http:", "https:"].includes(urlObj.protocol) ? url : urlObj.protocol,
-        recordingId,
-      });
-      return null;
-    }
 
-    return {
-      url,
-      text: await response.text(),
-    };
+    return await new Promise(resolve => {
+      NetUtil.asyncFetch(channel, function(inputStream, resultCode) {
+        if (resultCode !== 0) {
+          pingTelemetry("sourcemap-upload", "fetch-bad-status", {
+            message: `Request got result code: ${resultCode}`,
+            status: resultCode,
+            url: ["http:", "https:"].includes(urlObj.protocol) ? url : urlObj.protocol,
+            recordingId,
+          });
+
+          return null;
+        }
+
+        const str = NetUtil.readInputStreamToString(inputStream, inputStream.available());
+        inputStream.close();
+
+        resolve({
+          url,
+          text: str,
+        });
+      });
+    });
```

Comment:
Mostly a style nit, but I usually find that it's best to keep the body of a `new Promise` callback as small as possible for just the reason I mentioned about the `return null;` above. It's super easy to forget to call `resolve` and `return` instead. So you could for instance try:
```
const { channel, resultCode } = await new Promise(resolve => {
  NetUtil.asyncFetch(channel, (inputStream, resultCode) => resolve({ inputStream, resultCode }));
});
if (resultCode !== 0) {
  pingTelemetry("sourcemap-upload", "fetch-bad-status", {
    message: `Request got result code: ${resultCode}`,
    status: resultCode,
    url: ["http:", "https:"].includes(urlObj.protocol) ? url : urlObj.protocol,
    recordingId,
  });
  return null;
}

const str = NetUtil.readInputStreamToString(inputStream, inputStream.available());
inputStream.close();
return {
  url,
  text: str,
};
```